stubs/CoreImmutableClasses.phpstub

```diff
@@ -16,6 +16,27 @@ class DateTimeZone
     public function __construct(string $timezone) {}
 }
 
+/**
+ * @psalm-immutable
+ *
+ * @template-covariant From of string|DateTimeInterface
+ * @implements IteratorAggregate<int, DateTimeInterface>
+ */
+class DatePeriod implements IteratorAggregate
+{
+    const EXCLUDE_START_DATE = 1;
+    /**
+     * @param From $from
+     * @param (From is string ? 0|self::EXCLUDE_START_DATE : DateInterval) $interval_or_options
+     * @param (From is string ? never : DateTimeInterface|positive-int) $end_or_recurrences
+     * @param (From is string ? never : 0|self::EXCLUDE_START_DATE) $options
+     */
+    public function __construct($from, $interval_or_options = 0, $end_or_recurrences = 1, $options = 0) {}
```

Comment:
Let's use the actual parameter names that [PHP uses](https://github.com/php/php-src/blob/9dcb0bd1dc966a70cf6386ca9d49979dd417e0c5/ext/date/php_date_arginfo.h#L464-L469) instead of going off the documentation so that named parameters work: https://3v4l.org/3Rh8g

They like to make it difficult in these weird cases where they want to pretend it uses method overloading :roll_eyes:

Reply:
Done :)